src/mjolnir/restrictionbuilder.cc

```diff
@@ -519,9 +513,9 @@ void build(const std::string& complex_restriction_from_file,
                     res_way_ids.push_back(restriction.to());
                   }
 
-                  tmp_ids = GetGraphIds(currentNode, reader, tileid, lock, res_way_ids);
+                  tmp_ids = GetGraphIds(currentNode, reader, lock, res_way_ids);
 
-                  if (tmp_ids.size()) {
+                  if (tmp_ids.size() && tmp_ids.back().Tile_Base() == tile_id) {
```

Comment:
it happens if we split way into a few edges and not all edges are from the same tile. 
E.g. you have the following
```cpp
A -- B -- C
          | 
          D
```
There are cases that A-B-C is single osmway. And if the edges `AB` and `BC` are from the different tiles that code produced redundant restrictions: they are harmless but useless